### PR TITLE
bugfix use OR operator for merging bloom filter

### LINKED ISSUE #27: Bloom filter merge removes subscribed neighbors
### Expected Behavior

I send out 3 SUBSCRIBE messages followed by 3 bloom filter merge. Lets take a look on the first one. I start with an empty bloom filter and merge one of my neighbor's bloom filter. I print out `this.hasNeighborSubscribedTo('0f01020202')`. The empty bloom filter will retun false as expected.

First hop
```
{"level":"info","message":"merging neighbor's bloom filter with our own","timestamp":"2017-02-19T10:14:25.792Z"}
{"level":"info","message":"Bloomfilter local  to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}}}","timestamp":"2017-02-19T10:14:25.793Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: false","timestamp":"2017-02-19T10:14:25.793Z"}
{"level":"info","message":"Bloomfilter remote to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[80,151,217,180,236,168,106,166,24,96,180,208,12,233,128,120,225,88,96,0]}}}","timestamp":"2017-02-19T10:14:25.794Z"}
{"level":"info","message":"Bloomfilter local        merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[0,64,1,0,0,8,0,64,4,0,6,0,0,17,0,0,2,72,0,0]}}}","timestamp":"2017-02-19T10:14:25.794Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: true","timestamp":"2017-02-19T10:14:25.794Z"}
```

Second hop
```
{"level":"info","message":"Bloomfilter local  to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}}}","timestamp":"2017-02-19T10:14:25.795Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: true","timestamp":"2017-02-19T10:14:25.795Z"}
{"level":"info","message":"Bloomfilter remote to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[64,194,173,184,4,158,245,84,40,144,238,212,0,142,48,136,146,24,64,232]}}}","timestamp":"2017-02-19T10:14:25.800Z"}
{"level":"info","message":"Bloomfilter local        merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[80,151,217,180,236,168,106,166,24,96,180,208,12,233,128,120,225,88,96,0]}}}","timestamp":"2017-02-19T10:14:25.800Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: true","timestamp":"2017-02-19T10:14:25.800Z"}
```

### Actual Behavior

I would expect `this.hasNeighborSubscribedTo('0f01020202')` to be always true. It should never be false again. On merging the second neighbor's bloom filter it returns false. All the time it is false at the end I will random relay PUBLISH messages. If it is true at the ende I will relay PUBLISH to my neighbors.

`First hop
```
{"level":"info","message":"merging neighbor's bloom filter with our own","timestamp":"2017-02-19T10:14:25.836Z"}
{"level":"info","message":"Bloomfilter local  to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[0,64,1,0,0,8,0,64,4,0,6,0,0,17,0,0,2,72,0,0]}}}","timestamp":"2017-02-19T10:14:25.836Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: true","timestamp":"2017-02-19T10:14:25.836Z"}
{"level":"info","message":"Bloomfilter remote to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[80,151,217,180,236,168,106,166,24,96,180,208,12,233,128,120,225,88,96,0]}}}","timestamp":"2017-02-19T10:14:25.836Z"}
{"level":"info","message":"Bloomfilter local        merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[0,128,2,0,0,16,0,128,8,0,12,0,0,34,0,0,4,144,0,0]}}}","timestamp":"2017-02-19T10:14:25.837Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: false","timestamp":"2017-02-19T10:14:25.837Z"}
```

Second hop
```
{"level":"info","message":"Bloomfilter local  to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[80,151,217,180,236,168,106,166,24,96,180,208,12,233,128,120,225,88,96,0]}}}","timestamp":"2017-02-19T10:14:25.837Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: false","timestamp":"2017-02-19T10:14:25.837Z"}
{"level":"info","message":"Bloomfilter remote to be merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[64,194,173,184,4,158,245,84,40,144,238,212,0,142,48,136,146,24,64,232]}}}","timestamp":"2017-02-19T10:14:25.837Z"}
{"level":"info","message":"Bloomfilter local        merged: {\"size\":160,\"slices\":2,\"bitfield\":{\"buffer\":{\"type\":\"Buffer\",\"data\":[160,46,178,104,216,80,212,76,48,192,104,160,24,210,0,240,194,176,192,0]}}}","timestamp":"2017-02-19T10:14:25.837Z"}
{"level":"info","message":"Neighbors subscribed to 0f01020202: false","timestamp":"2017-02-19T10:14:25.837Z"}
```

### PR DESCRIPTION
**Pull request by littleskunk**
* Closes #27

### CODE REVIEW

File: lib/quasar.js [hunk 1]
```diff
@@ -177,11 +177,11 @@ Quasar.prototype._applyAttenuatedBloomFilterUpdate = function(atbf) {
   for (var f = 1; f < this._atbf.length; f++) {
     // Then for each bloom filter in our neighbor's response, merge their
     // bloom filter with ours for the given "hop" in our attenuated filter
-    for (var b = 0; b < atbf[f].bitfield.buffer.length; b++) {
+    for (var b = 0; b < atbf[f - 1].bitfield.buffer.length; b++) {
       var local = this._atbf[f].bitfield.buffer;
       var remote = atbf[f - 1].bitfield.buffer;
 
-      local[b] += remote[b];
+      local[b] = local[b] || remote[b];
```
Comment (bookchin) [inline lib/quasar.js]: Did you mean to use a bitwise OR (`|`) instead? This is a comparator.

Comment (bookchin) [inline lib/quasar.js]: Sorry, rather a logical or, not a comparator - it's early. This will return one value or the other.

Comment (littleskunk) [inline lib/quasar.js]: Thank you. Fixed and squashed.

**Comment by bookchin** [comment]
Will tag and publish when I get a moment today

**Pull request merged**